test/iht/views/GenericNonSubmittablePageBehaviour.scala

```diff
@@ -55,7 +55,11 @@ trait GenericNonSubmittablePageBehaviour extends ViewTestHelper {
       "show the exit link with the correct target and text" in {
         exitComponent.foreach { attrib =>
           val anchor = doc.getElementById(exitId)
-          anchor.attr("href") shouldBe attrib.target.url
+          if(attrib.hash > ""){
```

Comment:
Better, clearer, to use .nonEmpty here. Personally I'd probably write it like this:
          val expectedResult = if(attrib.hash.nonEmpty) attrib.target.url + attrib.hash else attrib.target.url
          cancelButton.attr("href") shouldBe expectedResult
but that's perhaps more a matter of personal taste.
Also, this bit of code is repeated 4 times throughout tests, so I'd put it somewhere common such as TestHelper.